src/request/actions/ActionCreator.ts

```diff
@@ -0,0 +1,32 @@
+import { Action } from 'redux';
+
+interface IActionCreatorAction<TPayload> extends Action {
+    payload: TPayload;
+}
+
+interface IActionCreatorBase<TPayload> {
+    type: string;
+    unwrap(action: Action): TPayload;
+}
+
+interface IActionCreator<TPayload> extends IActionCreatorBase<TPayload> {
+    (payload: TPayload): IActionCreatorAction<TPayload>;
+}
+
+export function createActionCreator<TPayload>(type: string): IActionCreator<TPayload> {
+    const actionCreator = (payload: TPayload) => {
+        return {
+            type: type,
+            payload: payload
```

Comment:
As a more general question, I wonder if this should be `{ type, ...payload}` instead of having payload hanging off a property?